### PR TITLE
Add env-based datasource config section

### PR DESCRIPTION
**Pull request by kjdelisle**
connected to https://github.com/strongloop-internal/scrum-loopback/issues/1396

If there's a section for this configuration already, I had great difficulty finding it.

Just adding a small bit about environment-based configuration, based on a customer question.

### CODE REVIEW

**Comment by crandmck** [comment]
Thanks for the PR.... We do have https://loopback.io/doc/en/lb2/Environment-specific-configuration.html#data-source-configuration

I think your addition would be better placed there.  Could you please move it there?  You could leave a link in `Advanced-topics-data-sources.html` if you like.

Also, since you had a hard time finding the, I'd also suggest adding cross-references to it wherever it  makes sense.

**Review by crandmck — CHANGES_REQUESTED** [review]
Please see my comments above.

File: pages/en/lb3/Environment-specific-configuration.md [hunk 1]
```diff
@@ -187,6 +187,22 @@ Example data sources:
...
+    connector: 'mongodb',
+    hostname: process.env.DB_HOST,
+    port: process.env.DB_PORT || 27017,
+    user: process.env.DB_USER,
+    password: process.env.DB_PASSWORD,
+    database: 'myapp',
+  }
+```
+
+
```
Comment (kjdelisle) [inline pages/en/lb3/Environment-specific-configuration.md:207]: Should I add that blurb about setting `export PRODUCTION=true` or `set PRODUCTION=true` here to make it clearer?

Comment (crandmck) [reply pages/en/lb3/Environment-specific-configuration.md:207]: Yes, I think that makes sense.

**Comment by crandmck** [comment]
Thanks!

**Pull request merged**